### ISSUE TITLE
UsdSkel display crash when changing a primitive type to SkelRoot

### ISSUE THREAD

**Issue by cameronwhite** [PixarAnimationStudios/USD#1248]
### Description of Issue

This is a stripped down version of a crash we encountered with UsdSkel in Houdini, and can reproduce in `usdview`.
I think this is a separate issue from the one reported in #1228 - it doesn't require the skeleton prim to be displayed, and has a different stack trace when it crashes

### Steps to Reproduce
1. Extract [usdskel_rebind.tar.gz](https://github.com/PixarAnimationStudios/USD/files/4794559/usdskel_rebind.tar.gz) and enter the folder
2. Copy over the HumanFemale folder from the UsdSkel sample, which is too large to include as a Github attachment. This should be copied twice to `HumanFemale_01` and `HumanFemale_02` - referencing different USD files seems to be required to replicate the sequence of changes that caused the crash in Houdini.
3. Run `usdview usdskel_xform.usda`
4. In the interpreter, run
```
l = Sdf.Layer.FindOrOpen("usdskel_skelroot.usda")
usdviewApi.stage.GetRootLayer().TransferContent(l)
```
5. This usually crashes immediately, but otherwise scrubbing the timeline should do the trick

### System Information (OS, Hardware)

Ubuntu 18.04, gcc 7.5

### Package Versions

USD 20.05 (https://github.com/sideeffects/USD/tree/dev_houdini18.5.197)

**Comment by jtran56** [comment]
Filed as internal issue #USD-6168.

**Comment by tallytalwar** [comment]
Thanks, this was included in the 21.02 release.
